### DISCUSSION TITLE
prevent transition feature

### DISCUSSION THREAD

**Discussion by diontools** [solidjs/solid-router#145] [Ideas]
I would like the ability to prevent browser backs. This would be used in conjunction with `beforeunload`.
Other libraries have similar functionality.

* [Next.js: beforePopState](https://nextjs.org/docs/api-reference/next/router#routerbeforepopstate)
* [Vue: navigation guards](https://router.vuejs.org/guide/advanced/navigation-guards.html)
* [history: block](https://github.com/remix-run/history/blob/main/docs/blocking-transitions.md)

**Comment by Brendan-csel** [comment]
Related [PR for `useBeforeLeave` hook](https://github.com/solidjs/solid-router/pull/186)
